src/GafferImage/Mix.cpp

```diff
@@ -186,6 +230,21 @@ void Mix::hashChannelData( const GafferImage::ImagePlug *output, const Gaffer::C
 {
 	ImageProcessor::hashChannelData( output, context, h );
 
+	float mix = mixPlug()->getValue();
+
+	if( mix == 0.0f )
+	{
+		h = inPlugs()->getChild< ImagePlug >( 0 )->channelDataPlug()->hash();
+		return;
+	}
+	else if( mix == 1.0f && !maskPlug()->getInput<ValuePlug>() )
+	{
+		h = inPlugs()->getChild< ImagePlug >( 1 )->channelDataPlug()->hash();
+		return;
+	}
+
+	h.append( mix );
```

Comment:
The `ImageProcessor::hashChannelData()` call should be here, just before the first `append()` - when we're making a direct pass through, there's no point initialising the hash because we're about to overwrite it.